### ISSUE TITLE
Cannot deobfuscate private fields

### ISSUE THREAD

**Issue by Kaizer69** [ben-sb/javascript-deobfuscator#29]
Hello! I try to deobfuscate a file that have private fields (hash char #):
[here docs on mozilla](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields)

```
..............node_modules\shift-parser\dist\tokenizer.js:1536
        throw this.createILLEGAL();
        ^
Error: [1:6801]: Unexpected "#"
Console error:
  index: 6800,
  line: 1,
  column: 6801,
  parseErrorLine: 1,
  parseErrorColumn: 6801,
  description: 'Unexpected "#"'
```

I guess it's a problem related to shift-parser that not support those new spec.

**Comment by ben-sb** [comment]
Yeah it is, only option to fix that would be rewriting the repo using another parser lib which I don't have time to do currently sadly

**Comment by 0x1Avram** [comment]
I, too, was uninspired when choosing a parser for a more targeted deobfuscator. I went with esprima, which doesn't support a lot of newer language features as well. As a workaround I found that transpiling the code to a compatible version via babel before deobfuscation works well and removes the need of rewriting with another parser.
